flos/special/dneb.lua

```diff
@@ -44,4 +45,143 @@ function DNEB:neb_force(image)
+      tmp[i] = self.k[i]
+   end
+   print("DNEB spring constant: ")
+   print(tostring(tmp))
+end
+
+
+function DNEB:force(image, IO)
+   self:_check_image(image)
+
+   if image == 1 then
+      -- Increment step-counter
+      self.niter = self.niter + 1
+   end
+
+   local NEB_F = self:neb_force(image)
+
+   -- Things I want to output in files as control (all in 3xN format)
+   if IO then
+      local f
+
+      -- Current coordinates (ie .R)
+      f = io.open( ("DNEB.%d.R"):format(image), "a")
+      self[image].R:savetxt(f)
+      f:close()
+
+      -- Forces before (ie .F)
+      f = io.open( ("DNEB.%d.F"):format(image), "a")
+      self[image].F:savetxt(f)
+      f:close()
+
+      -- Perpendicular force
+      f = io.open( ("DNEB.%d.F.P"):format(image), "a")
+      self:perpendicular_force(image):savetxt(f)
+      f:close()
+      
+      -- Spring force parallel
+      f = io.open( ("DNEB.%d.F.S.parallel"):format(image), "a")
+      self:spring_force(image):savetxt(f)
+      f:close()
+      
+      -- Spring force perdpendicular
+      f = io.open( ("DNEB.%d.F.S.perpendicular"):format(image), "a")
+      self:perpendicular_spring_force(image):savetxt(f)
+      f:close()
+
+      -- NEB Force
+      f = io.open( ("DNEB.%d.F.NEB"):format(image), "a")
+      NEB_F:savetxt(f)
+      f:close()
+
+      -- Tangent
+      f = io.open( ("DNEB.%d.T"):format(image), "a")
+      self:tangent(image):savetxt(f)
+      f:close()
+
+      -- dR - previous reaction coordinate
+      f = io.open( ("DNEB.%d.dR_prev"):format(image), "a")
+      self:dR(image-1, image):savetxt(f)
+      f:close()
+
+      -- dR - next reaction coordinate
+      f = io.open( ("DNEB.%d.dR_next"):format(image), "a")
+      self:dR(image, image+1):savetxt(f)
+      f:close()
+
+   end
+
+   -- Fake return to test
+   return NEB_F
+   
+end
+
+
+function DNEB:init_files()
+   
+   -- We clean all image data for a new run
+   local function new_file(fname, ...)
+      local f = io.open(fname, 'w')
+      local a = {...}
+      for _, v in pairs(a) do
+	 f:write("# " .. v .. "\n")
+      end
+      f:close()
+   end
+
+   new_file("DNEB.results", "DNEB results file",
+	    "Image reaction-coordinate Energy E-diff Curvature F-max(atom)")
+   
+   for img = 1, self.n_images do
+      new_file( ("DNEB.%d.R"):format(img), "Coordinates")
+      new_file( ("DNEB.%d.F"):format(img), "Constrained force")
+      new_file( ("DNEB.%d.F.P"):format(img), "Perpendicular force")
+      new_file( ("DNEB.%d.F.S.parallel"):format(img), "Spring parallel force")
+      new_file( ("DNEB.%d.F.S.perpendicular"):format(img), "Spring perpendicular force")
+      new_file( ("DNEB.%d.F.NEB"):format(img), "Resulting NEB force")
+      new_file( ("DNEB.%d.T"):format(img), "NEB tangent")
+      new_file( ("DNEB.%d.dR_prev"):format(img), "Reaction distance (previous)")
+      new_file( ("DNEB.%d.dR_next"):format(img), "Reaction distance (next)")
+   end
+
+end
+
+function DNEB:file_exists(name)--name
```

Comment:
Also, some of these should be needed? No, if you use inheritance all should be good?